spec/decorators/base_decorator_spec.rb

```diff
@@ -40,11 +44,31 @@
         User.all.to_a
       end
 
-      it 'calls decorate_collection with the collection' do
-        expect(Tramway::Decorators::CollectionDecorators).to receive(:decorate_collection)
-          .with(collection: relation, decorator:)
-        described_class.decorate(relation)
+      include_examples 'Decorate Collection'
+    end
+
+    context 'when object_or_array is an ActiveRecord::Relation by association' do
+      let(:relation) do
+        user = create :user
+        create_list(:post, 5, user:)
+
+        user.reload
+        user.posts
```

Comment:
why user.posts, couldn't you use just `create_list(:users)`?

Reply:
because `user.posts` is `Post::ActiveRecord_Associations_CollectionProxy` object. And I need this object for such test :slightly_smiling_face:

Reply:
it feels to me like there should be simplier way to get this, but can't propose a solution atm